src/views/Design.vue

```diff
@@ -0,0 +1,362 @@
+          </v-card>
+          <v-btn class="primary" :disabled="!isValid" @click="onSubmit"
+            >Submit
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+    <v-snackbar color="error" v-model="hasSubmissionError" :timeout="8000">
+      Sorry, submitting your design job failed. Please try again in a few
+      seconds or contact us if the problem persists.
+    </v-snackbar>
+  </v-container>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Route } from "vue-router";
+import axios, { AxiosResponse } from "axios";
+import * as settings from "@/utils/settings";
+import { OrganismItem } from "@/store/modules/organisms";
+import { ProjectItem } from "@/store/modules/projects";
+import { ModelItem, organism2ModelMapping } from "@/store/modules/models";
+import NewProject from "@/components/NewProject.vue";
+import NewOrganism from "@/components/NewOrganism.vue";
+import NewModel from "@/components/NewModel.vue";
+import { NextHandler } from "@/router";
+import { RuleHandler, RuleOutcome } from "@/types/forms";
+
+interface ProductItem {
+  name: string;
+}
+
+interface DesignState {
+  isValid: boolean;
+  project?: ProjectItem;
+  projectRules: ReadonlyArray<RuleHandler>;
+  isProjectCreationDialogVisible: boolean;
+  organism?: OrganismItem;
+  organismRules: ReadonlyArray<RuleHandler>;
+  isOrganismCreationDialogVisible: boolean;
+  product?: ProductItem;
+  productRules: ReadonlyArray<RuleHandler>;
+  productOptions: ProductItem[];
+  isLoadingProducts: boolean;
+  isAerobic: boolean;
+  showAdvanced: boolean;
+  bigg: boolean;
+  rhea: boolean;
+  model?: ModelItem;
+  isModelCreationDialogVisible: boolean;
+  maxPredictions: number;
+  predictionRules: ReadonlyArray<RuleHandler>;
+  isSubmitted: boolean;
+  hasSubmissionError: boolean;
+}
+
+export default Vue.extend({
+  name: "Design",
+  components: {
+    NewProject,
+    NewOrganism,
+    NewModel
+  },
+  data(): DesignState {
+    return {
+      isValid: false,
+      project: undefined,
+      projectRules: [v => !!v || "Project is required"],
+      isProjectCreationDialogVisible: false,
+      organism: undefined,
+      organismRules: [v => !!v || "Organism is required"],
+      isOrganismCreationDialogVisible: false,
+      product: undefined,
+      productRules: [v => !!v || "Product is required"],
+      productOptions: [],
+      isLoadingProducts: true,
+      isAerobic: false,
+      showAdvanced: false,
+      bigg: false,
+      rhea: true,
+      model: undefined,
+      isModelCreationDialogVisible: false,
+      maxPredictions: 3,
+      predictionRules: [
+        (v: number) => (v > 0 && v <= 10) || "Must be between 1 and 10"
+      ],
+      isSubmitted: false,
+      hasSubmissionError: false
+    };
+  },
+  computed: {
+    projectOptions(): ProjectItem[] {
+      return this.$store.state.projects.projects;
+    },
+    // TODO (Moritz Beber): Selection of organism in future should depend on the
```

Comment:
Opening an issue here will make corrections like this more manageable than TODO comments in the source code.

Reply:
#42